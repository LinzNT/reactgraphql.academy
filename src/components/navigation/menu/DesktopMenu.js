import PropTypes from 'prop-types'
import styled from 'styled-components'
import Link, { styleChildLinkColor } from '../../navigation/Link'
import React from 'react'
import DefaultUl, { Li } from '../../layout/Ul'
import MenuData from './Menu.json'
import { HideComponentsUsingCss } from '../../utils'
import { WHITE } from '../../../config/styles'

const Ul = styled(DefaultUl)`
  > li {
    margin: 0 !important;
    padding-left: 10px;
    padding-right: 12px;
    color: ${WHITE};
    ${styleChildLinkColor(WHITE)};
  }
`

const Item = ({ children, ...props }) => (
  <Li>
    <Link {...props}>{children}</Link>
  </Li>
)

export const DesktopMenuItem = styled(Item)`
  margin: 0;
`

DesktopMenuItem.displayName = 'DesktopMenuItem'

const DesktopMenu = () => (
  <HideComponentsUsingCss xs sm>
    <Ul inline>
      {MenuData.map((item, i) => (
        <DesktopMenuItem key={i} to={item.to}>
          {item.text}
        </DesktopMenuItem>
      ))}
    </Ul>
  </HideComponentsUsingCss>
)

export default DesktopMenu
