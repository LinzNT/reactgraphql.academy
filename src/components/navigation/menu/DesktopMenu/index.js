import PropTypes from 'prop-types'
import styled from 'styled-components'
import Link, { styleChildLinkColor } from '../../../navigation/Link'
import React from 'react'
import DefaultUl, { Li } from '../../../layout/Ul'
import MenuData from '../Menu.json'
import { HideComponentsUsingCss } from '../../../utils'
import { WHITE } from '../../../../config/styles'
import ItemSubmenu from './ItemSubmenu'

const Ul = styled(DefaultUl)`
  > li {
    margin: 0 !important;
    padding-left: 10px;
    padding-right: 12px;
    color: ${WHITE};
    ${styleChildLinkColor(WHITE)};
  }
`

class Item extends React.Component {
  render() {
    const { text, submenuItems, ...props } = this.props

    return (
      <Li>
        {submenuItems ? (
          <ItemSubmenu items={submenuItems} text={text} to={props.to} />
        ) : (
          <Link {...props}>{text}</Link>
        )}
      </Li>
    )
  }
}

export const DesktopMenuItem = styled(Item)`
  margin: 0 0 0 6px;
`
DesktopMenuItem.displayName = 'DesktopMenuItem'

const DesktopMenu = () => (
  <HideComponentsUsingCss xs sm>
    <Ul inline>
      {MenuData.map(item => (
        <DesktopMenuItem
          key={item.to}
          to={item.to}
          text={item.text}
          submenuItems={item.children}
        />
      ))}
    </Ul>
  </HideComponentsUsingCss>
)

export default DesktopMenu
