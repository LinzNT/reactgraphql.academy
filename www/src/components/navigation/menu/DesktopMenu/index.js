import styled from 'styled-components'
import Link, { styleChildLinkColor } from '../../../navigation/Link'
import React from 'react'
import DefaultUl, { Li } from '../../../layout/Ul'
import MenuData from '../Menu.json'
import { HideComponentsUsingCss } from '../../../utils'
import { WHITE } from '../../../../config/styles'
import ItemSubmenu from './ItemSubmenu'

const Ul = styled(DefaultUl)`
  > li {
    margin: 0 !important;
    padding-left: 10px;
    padding-right: 12px;
    color: ${WHITE};
    ${styleChildLinkColor(WHITE)};
    a {
      text-shadow: 0px 0px 1px ${WHITE};
    }
  }
`

class Item extends React.Component {
  render() {
    const { text, submenuItems, variant, variants, ...rest } = this.props

    return (
      <Li variant={variant} variants={variants}>
        {submenuItems ? (
          <ItemSubmenu
            className="navigation"
            items={submenuItems}
            text={text}
            to={rest.to}
            {...rest}
          />
        ) : (
          <Link {...rest}>{text}</Link>
        )}
      </Li>
    )
  }
}

export const DesktopMenuItem = styled(Item).attrs(props => ({
  className: 'navigation',
}))`
  margin: 0 0 0 6px;
`
DesktopMenuItem.displayName = 'DesktopMenuItem'

const DesktopMenu = () => (
  <HideComponentsUsingCss xs sm>
    <Ul variant="inline">
      {MenuData.map(item => (
        <DesktopMenuItem
          key={item.to}
          to={item.to}
          text={item.text}
          submenuItems={item.children}
        />
      ))}
    </Ul>
  </HideComponentsUsingCss>
)

export default DesktopMenu
