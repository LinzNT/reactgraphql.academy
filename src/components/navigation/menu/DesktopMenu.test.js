import React from 'react'
import { shallow } from 'enzyme'

import DesktopMenu, { DesktopMenuItem } from './DesktopMenu'

describe('<DesktopMenu />', () => {
    it('should render the list of menu items', () => {
        const data = [{ to: 'lorem', text: 'ipsum' }]

        const wrapper = shallow(<DesktopMenu data={data} />)

        expect(wrapper.find(DesktopMenuItem).length).toBe(1)
    })
})