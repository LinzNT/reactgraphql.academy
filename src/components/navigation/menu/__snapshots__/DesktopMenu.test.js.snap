// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DesktopMenu /> should render the list of menu items 1`] = `
<Styled(Components)
  sm={true}
  xs={true}
>
  <Styled(styled.ul)
    inline={true}
  >
    <DesktopMenuItem
      key="/curriculum"
      text="Curriculum"
      to="/curriculum"
    />
    <DesktopMenuItem
      key="/react-redux-graphql-bootcamp"
      submenuItems={
        Array [
          Object {
            "text": "Bootcamp",
            "to": "/react-redux-graphql-bootcamp",
          },
          Object {
            "text": "Corporate team training",
            "to": "/corporate-team-training",
          },
          Object {
            "text": "Advanced React",
            "to": "/advanced-react-redux-graphql-bootcamp",
          },
          Object {
            "text": "React Native",
            "to": "/react-native-bootcamp",
          },
          Object {
            "text": "Part-time",
            "to": "/react-redux-graphql-part-time-course",
          },
        ]
      }
      text="Training"
      to="/react-redux-graphql-bootcamp"
    />
    <DesktopMenuItem
      key="/community"
      text="Community"
      to="/community"
    />
    <DesktopMenuItem
      key="/about-us"
      text="About us"
      to="/about-us"
    />
    <DesktopMenuItem
      key="/blog"
      text="Blog"
      to="/blog"
    />
    <DesktopMenuItem
      key="#contact-us"
      text="Contact"
      to="#contact-us"
    />
  </Styled(styled.ul)>
</Styled(Components)>
`;
